tests/ctests/meson.build

```diff
@@ -1,3 +1,10 @@
+project('ctests', 'c')
```

Comment:
I'd like to get @daniloegea's opinion on this new project structure. Overall, it looks fine to me and I'm happy to change it that way, as long as we can run our tests as usual and don't need to adopt the packaging.

Reply:
The ctests were intentionally not linked against libnetplan. The idea was to build only the necessary files with the test files. The problem is that each file has dependencies spread across almost all the others so to get it working properly we need to do a good refactoring in the C code...

I think it should be fine to separate and link them against libnetplan to make things easier for the time being as long as we still can run them as part of `meson test` and the coverage checks still work.

Reply:
thought: would adding `link_with: libnetplan` to `tests/ctests/meson.build` work?